kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -69,6 +70,45 @@ public class TableConfig<T> {
      * All the valid properties that can be set on the table.
      */
     private static final HashMap<String, TableConfig> validProperties = new HashMap<>();
+
+    /**
+     * This table property is used to track the enablement of the inCommitTimestamps.
+     */
+    public static final TableConfig<Boolean> IN_COMMIT_TIMESTAMPS_ENABLED = new TableConfig<>(
+            "delta.enableInCommitTimestamps-preview",
+            "false",
+            Boolean::valueOf,
+            value -> true,
+            "needs to be a boolean."
+    );
+
+    /**
+     * This table property is used to track the version of the table at which inCommitTimestamps
+     * were enabled.
+     */
+    public static final TableConfig<Optional<Long>> IN_COMMIT_TIMESTAMP_ENABLEMENT_VERSION =
+            new TableConfig<>(
+                    "delta.inCommitTimestampEnablementVersion-preview",
+                    null,
+                    v -> Optional.ofNullable(v).map(Long::valueOf),
+                    value -> true,
+                    "needs to be a long."
+            );
+
+    /**
+     * This table property is used to track the timestamp at which inCommitTimestamps were enabled.
+     * More specifically, it is the inCommitTimestamp of the commit with the version specified in
+     * [[IN_COMMIT_TIMESTAMP_ENABLEMENT_VERSION]].
```

Comment:
`[[..]]` does this work in Javadoc? I thought it is only Scala doc. If it doesn't work use `{@link #..}`